lib/internal/cluster/RoundRobinHandle.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+const assert = require('assert');
+const net = require('net');
+const { sendHelper } = require('internal/cluster/utils');
+const uv = process.binding('uv');
+
+
+module.exports = RoundRobinHandle;
+
+
+function RoundRobinHandle(key, address, port, addressType, fd) {
+  this.key = key;
+  this.all = {};
+  this.free = [];
+  this.handles = [];
+  this.handle = null;
+  this.server = net.createServer(assert.fail);
+
+  if (fd >= 0)
+    this.server.listen({ fd: fd });
+  else if (port >= 0)
+    this.server.listen(port, address);
+  else
+    this.server.listen(address);  // UNIX socket path.
+
+  this.server.once('listening', () => {
+    this.handle = this.server._handle;
+    this.handle.onconnection = (err, handle) => this.distribute(err, handle);
+    this.server._handle = null;
+    this.server = null;
+  });
```

Comment:
While we're at it, perhaps consider eliminating the closure by moving these handlers into top level functions. Doing so should result in at least some performance improvement overall.